### ISSUE TITLE
Spurious clang-cl warnings compiling atomics for ARM64EC

### ISSUE THREAD

**Issue by skoulik** [llvm/llvm-project#159229]
The problem is only reproducible when compiling ARM64EC targets. Using std::atomic_bool::load, std::atomic_signal_fence and others produce warnings: '+cx16' is not a recognized feature for this target (ignoring feature)

Compiling for ARM64, AMD64 does not produce the warnings.

clang-cl --version
clang version 21.1.0
Target: x86_64-pc-windows-msvc
Thread model: posix

test.cpp:
#include <atomic>
int main()
{
    std::atomic_signal_fence(std::memory_order_seq_cst);
    return 0;
}

test.cpp (another example):
#include <atomic>
int main()
{
    std::atomic_bool test {false};
    auto t = test.load(std::memory_order_seq_cst);
    return 0;
}

clang-cl --target=arm64ec-pc-windows-msvc test.cpp
'+cx16' is not a recognized feature for this target (ignoring feature)
'+cx16' is not a recognized feature for this target (ignoring feature)

clang-cl --target=arm64-pc-windows-msvc test.cpp
(empty output)

clang-cl --target=amd64-pc-windows-msvc test.cpp
(empty output)

**Comment by efriedma-quic** [comment]
Reduced:

```
#pragma clang attribute _STD_ATOMIC_HEADER.push([[gnu::target("cx16")]], apply_to = function)
extern "C" inline void atomic_signal_fence() noexcept {}
#pragma clang attribute _STD_ATOMIC_HEADER.pop

int main()
{
  atomic_signal_fence();
  return 0;
}
```

There is a clang bug here: the frontend should print a proper error/warning instead of just passing the raw feature flag to the backend.  But the STL headers shouldn't do this.

It looks like there's a fix in the latest STL (https://github.com/microsoft/STL/pull/5717); CC @StephanTLavavej .
